config/cmake/hdf5-config.cmake.in

```diff
@@ -29,6 +29,8 @@ set (${HDF5_PACKAGE_NAME}_VALID_COMPONENTS
     Tools
 )
 
+set (${HDF5_PACKAGE_NAME}_BUILD_MODE       @HDF_CFG_NAME@)
```

Comment:
What's the motivation for introducing this new variable in addition to the pre-existing HDF_CFG_NAME?

Reply:
This records the setting to supply the use of the binary.